clinvar_ingest/cloud/bigquery/create_tables.py

```diff
@@ -48,36 +52,86 @@ def ensure_dataset_exists(
     return dataset
 
 
-def run_create(args):
+def schema_file_path_for_table(table_name: str) -> dict:
+    """
+    Loads the schema for the given table name.
+    """
+    base_path = Path("clinvar_ingest/cloud/bigquery/bq_json_schemas")
+    schema_path = base_path / f"{table_name}.bq.json"
+    return schema_path
+
+
+def create_table(
+    table_name: str,
+    dataset: bigquery.Dataset,
+    blob_uri: str,
+    client: bigquery.Client,
+) -> bigquery.Table:
+    """
+    Creates a table in the given dataset, using the given bucket and path.
+    """
+    table_ref = dataset.table(table_name)
+    external_config = bigquery.ExternalConfig(
+        bigquery.SourceFormat.NEWLINE_DELIMITED_JSON
+    )
+    external_config.source_uris = [blob_uri]
+    external_config.autodetect = True
+    external_config.schema = client.schema_from_json(
+        schema_file_path_for_table(table_name)
+    )
+
+    table = bigquery.Table(table_ref, schema=None)
+    table.external_data_configuration = external_config
+    table = client.create_table(table, exists_ok=True)
+    return table
+
+
+def run_create(args: CreateExternalTablesRequest) -> dict[str, bigquery.Table]:
```

Comment:
could this be renamed to `run_create_external_tables`?

Reply:
i think that makes sense to me. It's interesting because I've been using regular "Tables" in the current state of the system (I copy the tdr views into my own tables) so i don't have "external tables". I'm wondering if there's going to be an issue with that (performance or feature wise). I'm assuming not, but we can always make a change later if necessary. I'd be curious to know if terra uses external tables or if they stick with standard sql tables for it's backend. probably GCS external tables?

Reply:
yes good idea, it's ambiguous when it's imported elsewhere

Reply:
@larrybabb I can't easily tell what the tables behind those TDR views are. I don't have permission to view them or get their details. I would assume they are native bigquery tables, not external tables.

If we are going to be doing a lot with the external tables that involves looking at specific columns, JSON will be worse computationally and bigger file sizes than something like Parquet. But I think since we are intending to use these as sort of temporary tables anyways, using JSON is fine.